lib/converter/policy.go

```diff
@@ -73,6 +99,15 @@ func (p PolicyConverter) ConvertKVPairToAPI(d *model.KVPair) (unversioned.Resour
 	ap.Spec.Selector = bp.Selector
 	ap.Spec.DoNotTrack = bp.DoNotTrack
 	ap.Spec.PreDNAT = bp.PreDNAT
+	ap.Spec.Types = nil
+
+	// Note: avoid accidentally converting from nil to [].
+	if bp.Types != nil {
```

Comment:
Does this mean there's a case where `ap.Spec.Types` will be `nil` on read?

I think we probably want to be populating the `Types` field with the correct values if they're not there when someone does a "get", since we're enforcing the presence of this field on "create" (via defaults and validation).

I was imagining this would be done in the etcd/kdd backends so that the defaulting would also apply to the Syncer.